tests/main/interfaces-ros-snapd-support/task.yaml

```diff
@@ -45,7 +45,10 @@ execute: |
     gojq . < response.txt
 
     echo "Check if the service is now disabled"
-    api-apps-client > response.txt
-    gojq . < response.txt
-
-    gojq -r '.result[] | select(.snap == "test-snapd-simple-service").enabled' < response.txt | MATCH '^null'
+    check_disabled() {
+        api-apps-client > response.txt
+        gojq . < response.txt
+        gojq -r '.result[] | select(.snap == "test-snapd-simple-service").enabled' < response.txt | MATCH '^null'
+    }
+    export -f check_disabled
+    retry -n 5 --wait 1 bash -c "check_disabled"
```

Comment:
Based on the search results https://github.com/search?q=repo%3Acanonical%2Fsnapd%20retry%20-n%20--wait%201&type=code I would propose something more like "-n 30"
Thanks for adding the retry semantic here, I didn't notice it was used in other tests when we made the original PR.

Reply:
@maykathm can we retry the api-apps-client and not the whole set of operations?

This would also not require bash to inherit the function definition.